wallet-new/src/Cardano/Wallet/Kernel/DB/AcidState.hs

```diff
@@ -123,21 +123,44 @@ deriveSafeCopy 1 'base ''RollbackDuringRestoration
   Wrap wallet spec
 -------------------------------------------------------------------------------}
 
+
+-- | Create a new pending transaction
+--
+--  NOTE: also creates all "our" addresses that appear in the transaction outputs,
+--        by adding them to the HdAddresses they will be henceforth recognised as "ours"
 newPending :: HdAccountId
            -> InDb Txp.TxAux
+           -> [AddrWithId] -- ^ "our" addresses that appear in the transaction outputs
```

Comment:
Let's not have this `AddrWithId` here in the acid-state, let's just take a `[HdAddress]`.